### PR TITLE
Fix memory issues found by coverity

### PR DESCRIPTION
**Pull request by kgaillot**
Also clean up the remote CIB authentication code

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , can you review this one when you get a chance? Thanks

**Comment by kgaillot** [comment]
Everything but the "case-sensitively" and "reject" commits will be backported to 2.1

File: daemons/based/based_remote.c [hunk 1]
```diff
@@ -617,40 +625,45 @@ authenticate_user(const char *user, const char *passwd)
...
+    pass = true;
 
-  bail:
+bail:
     pam_end(pam_h, rc);
-#endif
     return pass;
+#else
+    // @TODO Implement for non-PAM environments
+    return true;
```
Comment (clumens) [inline daemons/based/based_remote.c]: Just curious... what non-PAM environments do we support?

File: daemons/based/based_remote.c [hunk 2]
```diff
@@ -673,6 +670,8 @@ authenticate_user(const char *user, const char *passwd)
     return pass;
 #else
     // @TODO Implement for non-PAM environments
-    return true;
+    crm_err("Rejecting remote user %s because this build does not have "
+            "PAM support");
```
Comment (clumens) [inline daemons/based/based_remote.c:675]: Missing the string substitution here.

File: daemons/controld/controld_schedulerd.c [hunk 3]
```diff
@@ -494,11 +494,11 @@ do_pe_invoke_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, void
         crm_xml_add_int(output, PCMK_XA_NO_QUORUM_PANIC, 1);
     }
 
-    rc = pcmk_rc2legacy(pcmk_schedulerd_api_graph(schedulerd_api, output, &ref));
-
-    if (rc < 0) {
+    rc = pcmk_schedulerd_api_graph(schedulerd_api, output, &ref);
+    if (rc != pcmk_rc_ok) {
+        free(ref);
```
Comment (clumens) [inline daemons/controld/controld_schedulerd.c:499]: It's possible for pcmk_schedulerd_api_graph to return an error before ref has been allocated - for instance, if it returns ENOTCONN.

**Review by clumens — CHANGES_REQUESTED** [review]

File: daemons/based/based_remote.c (hunk 1 above)
Comment (kgaillot) [reply daemons/based/based_remote.c]: *BSD can install OpenPAM but don't have to

File: daemons/based/based_remote.c (hunk 2 above)
Comment (kgaillot) [reply daemons/based/based_remote.c]: whoops, thanks

updated commit

File: daemons/controld/controld_schedulerd.c (hunk 3 above)
Comment (kgaillot) [reply daemons/controld/controld_schedulerd.c:499]: yep, but free(NULL) is fine

**Comment by kgaillot** [comment]
Rebased on current main and added "empty variant variables" commit

**Review by clumens — APPROVED** [review]

**Comment by kgaillot** [comment]
@clumens, after testing, this changed significantly, so it needs a re-review

All the remote stuff turned out to be an unnecessary diversion, but at least that code is cleaner now ...

**Review by clumens — APPROVED** [review]
Looks good to me.

**Pull request merged**